tests/tensor/rewriting/test_linalg.py

```diff
@@ -906,3 +906,29 @@ def test_rewrite_cholesky_diag_to_sqrt_diag_not_applied():
     f_rewritten = function([x], z_cholesky, mode="FAST_RUN")
     nodes = f_rewritten.maker.fgraph.apply_nodes
     assert any(isinstance(node.op, Cholesky) for node in nodes)
+
+
+def test_dot_kron_rewrite():
```

Comment:
Add test with batch dims. I am worried the use of e.g. `ravel` in the rewrite will cause problems.